src/libfetchers/fetchers.cc

```diff
@@ -131,6 +131,12 @@ std::pair<Tree, Input> Input::fetch(ref<Store> store) const
     }
 
     auto [storePath, input] = [&]() -> std::pair<StorePath, Input> {
+        if (settings.offline) {
+            auto url = this->toURL();
+            if (url.scheme != "file" && url.scheme != "git+file")
```

Comment:
There are more input types that can work offline (at least `path` and `hg+file` come to my mind).
It’d probably be better to defer this check to the fetcher − maybe have a `Fetcher::isLocal(std::string_view url)` method that we could call here, or just lift this check inside `Fetcher::fetch`

Reply:
I tried putting the check in different places but decided doing it this dumb way was a reasonable baseline before modifiying multiple classes. I can move this check into `Fetcher`.

Reply:
Agreed with @thufschmitt